### DISCUSSION TITLE
config.json issue?

### DISCUSSION THREAD

**Discussion by fthrush** [devcompl/TTController#205] [Configuration]
I'm having trouble getting my config to play nice with my hardware it seems. My config file is as follows: 

> {
  "Profiles": [
    {
      "Name": "Default",
      "Ports": [
        [9802, 8102, 1],
        [9802, 8102, 2],
        [9802, 8102, 3],
        [9802, 8102, 4],
        [9802, 8102, 5]
      ],
      "SpeedControllers": [
        {
          "Type": "PwmSpeedController",
          "Config": {
            "CurvePoints": [
              [25, 35],
              [40, 40],
              [50, 50],
              [60, 60],
              [70, 80],
              [75, 100]
            ],
            "Sensors": ["/gpu-nvidia/0/temperature/0"],
            "MinimumChange": 4,
            "MaximumChange": 8,
            "Trigger": {
              "Type": "AlwaysTrigger"
            }
          }
        }
      ]
    }
  ],
  "ComputerStateProfiles": [
    {
      "StateType": "Boot",
      "Ports": [
        [9802, 8102, 1],
        [9802, 8102, 2],
        [9802, 8102, 3],
        [9802, 8102, 4],
        [9802, 8102, 5]
      ],
      "Speed": 35
    },
    {
      "StateType": "Shutdown",
      "Ports": [
        [9802, 8102, 1],
        [9802, 8102, 2],
        [9802, 8102, 3],
        [9802, 8102, 4],
        [9802, 8102, 5]
      ],
      "Speed": 35
    }
  ],
  "PortConfigs": [],
  "SensorConfigs": []
}

However when I run the service in the console, this is what i am seeing for portId which doesn't make a whole lot of sense to me:

> 19:50:01.2365: Port [9802, 8102, 1] data: [PortId: 252, Speed: 0, Rpm: 0, Unknown: 0]
19:50:01.2365: Port [9802, 8102, 2] data: [PortId: 252, Speed: 0, Rpm: 0, Unknown: 0]
19:50:01.2365: Port [9802, 8102, 3] data: [PortId: 252, Speed: 0, Rpm: 0, Unknown: 0]
19:50:01.2365: Port [9802, 8102, 4] data: [PortId: 252, Speed: 0, Rpm: 0, Unknown: 0]
19:50:01.2365: Port [9802, 8102, 5] data: [PortId: 252, Speed: 0, Rpm: 0, Unknown: 0]
19:50:01.2365: Sensor "/gpu-nvidia/0/temperature/0" value: 30.01891

And for sanities sake, below is the controller information:

> »  Name: Riing Plus
»        VendorId: 9802
»        ProductId: 8102
»        Version: 2.0.3
»        Ports:
»                1:
»                        Data: [PortId: 1, Speed: 35, Rpm: 0, Unknown: 255]
»                        Identifier: [9802, 8102, 1]
»                2:
»                        Data: [PortId: 2, Speed: 35, Rpm: 0, Unknown: 255]
»                        Identifier: [9802, 8102, 2]
»                3:
»                        Data: [PortId: 3, Speed: 35, Rpm: 0, Unknown: 255]
»                        Identifier: [9802, 8102, 3]
»                4:
»                        Data: [PortId: 4, Speed: 35, Rpm: 483, Unknown: 255]
»                        Identifier: [9802, 8102, 4]
»                5:
»                        Data: [PortId: 5, Speed: 35, Rpm: 500, Unknown: 255]
»                        Identifier: [9802, 8102, 5]
»        Available effect types: Flow_Extreme, Flow_Fast, Flow_Normal, Flow_Slow, Spectrum_Extreme, Spectrum_Fast, Spectrum_Normal, Spectrum_Slow, Ripple_Extreme, Ripple_Fast, Ripple_Normal, Ripple_Slow, Blink_Extreme, Blink_Fast, Blink_Normal, Blink_Slow, Pulse_Extreme, Pulse_Fast, Pulse_Normal, Pulse_Slow, Wave_Extreme, Wave_Fast, Wave_Normal, Wave_Slow, PerLed, Full

I'm hoping i'm just making a stupid mistake here and someone smarter than I can enlighten me. Thanks!!

**Comment by MoshiMoshi0** [comment]
You might have other applications running which are trying to read data (RPM) from the fans, like AIDA or thermaltake software.
Also you might have to set [PortConfigs](https://moshimoshi0.github.io/TTController/config/service#portconfigs).

**Comment by Monanasa** [comment]
its true
